src/dune_engine/pform.mli

```diff
@@ -0,0 +1,129 @@
+open! Stdune
+
+module Var : sig
+  type t =
+    | User_var of string
+    | Nothing
+    | Project_root
+    | Workspace_root
+    | First_dep
+    | Deps
+    | Targets
+    | Target
+    | Cc
+    | Cxx
+    | Ccomp_type
+    | Cpp
+    | Pa_cpp
+    | Make
+    | Ocaml_version
+    | Ocaml
+    | Ocamlc
+    | Ocamlopt
+    | Ocaml_bin_dir
+    | Ocaml_stdlib_dir
+    | Dev_null
+    | Ext_obj
+    | Ext_asm
+    | Ext_lib
+    | Ext_dll
+    | Ext_exe
+    | Ext_plugin
+    | Profile
+    | Context_name
+    | Os_type
+    | Architecture
+    | Arch_sixtyfour
+    | System
+    | Model
+    | Ignoring_promoted_rules
+    | Input_file
+
+  val compare : t -> t -> Ordering.t
+
+  val to_dyn : t -> Dyn.t
+end
+
+module Artifact : sig
+  type t =
+    | Mod of Cm_kind.t
+    | Lib of Mode.t
+
+  val compare : t -> t -> Ordering.t
+
+  val to_dyn : t -> Dyn.t
+
+  val ext : t -> string
+
+  val all : t list
+end
+
+module Macro : sig
+  type t =
+    | Exe
+    | Dep
+    | Bin
+    (* All four combinations are allowed and correspond to variables [lib],
+       [libexec], [lib-private], and [libexec-private]. *)
+    | Lib of
+        { lib_exec : bool
+        ; lib_private : bool
+        }
+    | Lib_available
+    | Version
+    | Read
+    | Read_strings
+    | Read_lines
+    | Path_no_dep
+    | Ocaml_config
+    | Env
+    | Artifact of Artifact.t
+
+  val compare : t -> t -> Ordering.t
+
+  val to_dyn : t -> Dyn.t
+end
+
+type t =
+  | Var of Var.t
+  | Macro of Macro.t * string
+
+val compare : t -> t -> Ordering.t
+
+val to_dyn : t -> Dyn.t
+
+(** Enode using the latest name for a given percent form. Return [None] if the
+    percent form was deleted. *)
+val encode_to_latest_dune_lang_version : t -> (string * string option) option
```

Comment:
IMO this would be cleaner if there was a dedicated variant for the return value.

Reply:
Agreed, changing that